src/merkle/tiered_smt/proof.rs

```diff
@@ -110,6 +111,20 @@ impl TieredSmtProof {
             .expect("failed to compute Merkle path root")
     }
 
+    /// Consume the proof and returns its parts.
+    pub fn into_parts(self) -> (MerklePath, Vec<(RpoDigest, Word)>) {
+        (self.path, self.entries)
+    }
+
+    /// Create a proof from its parts
+    pub fn from_parts<I>(path: MerklePath, entries: I) -> Self
```

Comment:
Could we modify the constructor to take  a `MerklePath` and an iterator as we have here instead of introducing this additional method?